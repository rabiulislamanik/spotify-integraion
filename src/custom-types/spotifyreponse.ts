export interface ISpotifyCommonResponse {
    next:string|null,
    total:number
}

export interface IAlbums extends ISpotifyCommonResponse{
    items:IAlbumItem[]
}
  
export interface INewReleaseAlbumResponse{
    albums:IAlbums
}

export interface IAlbumItem {
    album_type:string,
    artists:{external_urls:{spotify:string},name:string}[],
    external_urls:{ spotify: string},
    name:string,
    release_date:string,
    total_tracks:number,
    images:{url:string}[]
}

export interface IPlaylists extends ISpotifyCommonResponse{
    items:IPlaylistItem[]
}

export interface IFeaturedPlaylistsResponse{
    message:string,
    playlists:IPlaylists
}


export interface IPlaylistItem {
    description:string,
    external_urls:{ spotify: string},
    name:string,
    owner:{display_name:string,external_urls:{ spotify: string}},
    images:{url:string}[]
}


export interface ICategories extends ISpotifyCommonResponse{
    items:ICategoryItem[]
}

export interface ICategoriesResponse{
    message:string,
    categories:ICategories
}

export interface ICategoryItem {
    icons:{url:string}[],
    name:string,
    id:string
}
