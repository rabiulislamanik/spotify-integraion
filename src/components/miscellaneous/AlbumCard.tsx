import React from "react";
import { IAlbumItem } from '../../custom-types/spotifyreponse';

interface IAlBumCard{
    album:IAlbumItem
}

function AlbumCard (props:IAlBumCard){
  return (
    <div className="card">
      <img className="card-image" src={props.album.images[1].url} alt="icon" height="128" width="128"/>
      <p>Album : <a className="card-link" href={props.album.external_urls.spotify}>{props.album.name}</a></p>
      <p>Album Type : {props.album.album_type}</p>
      <p>Total Tracks : {props.album.total_tracks}</p>
      <p>Artists : {props.album.artists.map((artist,i)=>
        (<a className="card-link" href={artist.external_urls.spotify} key={artist.name}>
          {artist.name+ " "}</a>
        ))}
      </p>
      <p>Released On : {props.album.release_date}</p>
    </div>
  );
}

export default AlbumCard;