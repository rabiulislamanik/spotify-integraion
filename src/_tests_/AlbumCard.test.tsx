import React from 'react';
import renderer from 'react-test-renderer';

import AlbumCard from '../components/miscellaneous/AlbumCard';
import { IAlbumItem } from '../custom-types/spotifyreponse';

it('renders correctly when an album is passed as prop', () => {
  const fakeAlbum:IAlbumItem = {
      album_type:'single',
      artists:[{external_urls:{spotify:'artist_fake_url'},name:"Fake Artist"}],
      external_urls:{ spotify: 'spotify_url'},
      name:'Album',
      release_date:'2020-11-11',
      total_tracks:1,
      images:[{url:'image_url1'},{url:'image_url2'},{url:'image_url3'}]   
     
  }; 
  const tree = renderer.create(<AlbumCard album={fakeAlbum}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

