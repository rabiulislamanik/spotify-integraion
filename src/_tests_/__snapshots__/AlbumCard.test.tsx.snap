// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when an album is passed as prop 1`] = `
<div
  className="card"
>
  <img
    alt="icon"
    className="card-image"
    height="128"
    src="image_url2"
    width="128"
  />
  <p>
    Album : 
    <a
      className="card-link"
      href="spotify_url"
    >
      Album
    </a>
  </p>
  <p>
    Album Type : 
    single
  </p>
  <p>
    Total Tracks : 
    1
  </p>
  <p>
    Artists : 
    <a
      className="card-link"
      href="artist_fake_url"
    >
      Fake Artist 
    </a>
  </p>
  <p>
    Released On : 
    2020-11-11
  </p>
</div>
`;
